core/commands/swarm.go

```diff
@@ -130,6 +134,30 @@ var swarmPeeringAddCmd = &cmds.Command{
 				return err
 			}
 		}
+
+		save, _ := req.Options[swarmSaveOptionName].(bool)
+		if save {
+			r, err := fsrepo.Open(env.(*commands.Context).ConfigRoot)
+			if err != nil {
+				return err
+			}
+			defer r.Close()
+			cfg, err := r.Config()
+			if err != nil {
+				return err
+			}
+
+			addrInfos, err := mergeAddrInfo(addInfos, cfg.Peering.Peers)
```

Comment:
Correct me if I'm wrong, but we can do the same as we are doing in line 248 to set the peers, avoiding even more duplicated lines.

Reply:
Good catch! Simplified in [this commit](https://github.com/ipfs/kubo/pull/9192/commits/94fc792df4ab6b858b4b9e980aa6f7ffca24bd03).